crates/fiber/src/unix/aarch64.rs

```diff
@@ -0,0 +1,177 @@
+//   `DW_CFA_AARCH64_negate_ra_state` DWARF operation (aliased with the
+//   `.cfi_window_save` assembler directive) informs an unwinder about this
+
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "macos")] {
+        macro_rules! cfi_window_save { () => (); }
+        macro_rules! pacia1716 { () => (); }
+        macro_rules! paciasp { () => (); }
+        macro_rules! autiasp { () => (); }
+    } else if #[cfg(target_os = "linux")] {
+        macro_rules! cfi_window_save { () => (".cfi_window_save\n"); }
+        macro_rules! pacia1716 { () => ("pacia1716\n"); }
+        macro_rules! paciasp { () => ("paciasp\n"); }
+        macro_rules! autiasp { () => ("autiasp\n"); }
+    } else {
+        compile_error!("unsupported aarch64 platform");
+    }
+}
+
+// fn(top_of_stack(%x0): *mut u8)
+asm_func!(
+    "wasmtime_fiber_switch",
+    "
+        .cfi_startproc
+    ",
+    paciasp!(),
+    cfi_window_save!(),
+    "
+        // Save all callee-saved registers on the stack since we're
+        // assuming they're clobbered as a result of the stack switch.
+        stp x29, x30, [sp, -16]!
+        stp x20, x19, [sp, -16]!
+        stp x22, x21, [sp, -16]!
+        stp x24, x23, [sp, -16]!
+        stp x26, x25, [sp, -16]!
+        stp x28, x27, [sp, -16]!
+        stp d9, d8, [sp, -16]!
+        stp d11, d10, [sp, -16]!
+        stp d13, d12, [sp, -16]!
+        stp d15, d14, [sp, -16]!
+
+        // Load our previously saved stack pointer to resume to, and save
+        // off our current stack pointer on where to come back to
+        // eventually.
+        ldr x8, [x0, -0x10]
+        mov x9, sp
+        str x9, [x0, -0x10]
+
+        // Switch to the new stack and restore all our callee-saved
+        // registers after the switch and return to our new stack.
+        mov sp, x8
+        ldp d15, d14, [sp], 16
+        ldp d13, d12, [sp], 16
+        ldp d11, d10, [sp], 16
+        ldp d9, d8, [sp], 16
+        ldp x28, x27, [sp], 16
+        ldp x26, x25, [sp], 16
+        ldp x24, x23, [sp], 16
+        ldp x22, x21, [sp], 16
+        ldp x20, x19, [sp], 16
+        ldp x29, x30, [sp], 16
+    ",
+    autiasp!(),
+    cfi_window_save!(),
+    "
+        ret
+        .cfi_endproc
+    ",
+);
+
+// fn(
+//    top_of_stack(%x0): *mut u8,
+//    entry_point(%x1): extern fn(*mut u8, *mut u8),
+//    entry_arg0(%x2): *mut u8,
+// )
+// We set up the newly initialized fiber, so that it resumes execution
+// from wasmtime_fiber_start(). As a result, we need a signed address
+// of this function, so there are 2 requirements:
+// * The fiber stack pointer value that is used by the signing operation
+//   must match the value when the pointer is authenticated inside
+//   wasmtime_fiber_switch(), otherwise the latter would fault
+// * We would like to use an instruction that is executed as a no-op by
+//   processors that do not support PAuth, so that the code is
+//   backward- compatible and there is no duplication; `PACIA1716` is a
```

Comment:
Nit - there is a space between `backward-` and `compatible`; it just looks weird.